src/steppy/base.py

```diff
@@ -0,0 +1,327 @@
+                adapted_steps[adapted_name] = step_inputs[mapping]
+            else:
+                if len(mapping) == 2:
+                    (step_mapping, func) = mapping
+                elif len(mapping) == 1:
+                    step_mapping = mapping
+                    func = identity_inputs
+                else:
+                    raise ValueError('wrong mapping specified')
+
+                raw_inputs = [step_inputs[step_name][step_var] for step_name, step_var in step_mapping]
+                adapted_steps[adapted_name] = func(raw_inputs)
+        return adapted_steps
+
+    def unpack(self, step_inputs):
+        logger.info('step {} unpacking inputs'.format(self.name))
+        unpacked_steps = {}
+        for step_name, step_dict in step_inputs.items():
+            unpacked_steps = {**unpacked_steps, **step_dict}
+        return unpacked_steps
+
+    @property
+    def all_steps(self):
+        all_steps = {}
+        all_steps = self._get_steps(all_steps)
+        return all_steps
+
+    def _get_steps(self, all_steps):
+        for input_step in self.input_steps:
+            all_steps = input_step._get_steps(all_steps)
+        all_steps[self.name] = self
+        return all_steps
+
+    @property
+    def graph_info(self):
+        graph_info = {'edges': set(),
+                      'nodes': set()}
+
+        graph_info = self._get_graph_info(graph_info)
+
+        return graph_info
+
+    def _get_graph_info(self, graph_info):
+        for input_step in self.input_steps:
+            graph_info = input_step._get_graph_info(graph_info)
+            graph_info['edges'].add((input_step.name, self.name))
+        graph_info['nodes'].add(self.name)
+        for input_data in self.input_data:
+            graph_info['nodes'].add(input_data)
+            graph_info['edges'].add((input_data, self.name))
+        return graph_info
+
+    def plot_graph(self, filepath):
+        plot_graph(self.graph_info, filepath)
+
+    def __str__(self):
+        return pprint.pformat(self.graph_info)
+
+    def _repr_html_(self):
+        return view_graph(self.graph_info)
+
+
+class BaseTransformer:
+    def fit(self, *args, **kwargs):
+        return self
+
+    def transform(self, *args, **kwargs):
+        return NotImplementedError
+
+    def fit_transform(self, *args, **kwargs):
+        self.fit(*args, **kwargs)
+        return self.transform(*args, **kwargs)
+
+    def load(self, filepath):
+        return self
+
+    def save(self, filepath):
+        joblib.dump({}, filepath)
+
+
+class MockTransformer(BaseTransformer):
+    def fit(self, *args, **kwargs):
+        return self
+
+    def transform(self, *args, **kwargs):
+        return
+
+    def fit_transform(self, *args, **kwargs):
+        self.fit(*args, **kwargs)
+        return self.transform(*args, **kwargs)
+
+
+class Dummy(BaseTransformer):
+    def transform(self, **kwargs):
+        return kwargs
+
+
+def to_tuple_inputs(inputs):
```

Comment:
@taraspiotr all of those adapter functions are not existent in steppy anymore. You simply create a step for that